src/lib/math.js

```diff
@@ -358,8 +358,46 @@ const $builtinmodule = function (name) {
         }
     };
 
-    function isqrt(x) {
-        throw new Sk.builtin.NotImplementedError("math.isqrt() is not yet implemented in Skulpt");
+    function bigint_isqrt(n_js) {
+        // Algorithm taken from:
+        // https://github.com/python/cpython/blob/main/Modules/mathmodule.c#L1598-L1618
+        let c = n_js.toString(2).length; // bitlength of a positive bigint
+        c = Math.floor((c - 1) / 2);
+        let s = c.toString(2).length; // bitlength
+        const one = JSBI.BigInt(1);
+        const two = JSBI.BigInt(2);
+        const cBigInt = JSBI.BigInt(c);
+        const twoC = JSBI.multiply(two, cBigInt);
+        let a = one;
+        let d = JSBI.BigInt(0);
+        while (s > 0) {
+            s--;
+            e = d;
```

Comment:
I'm not sure `e` has been declared. Likewise `n_js` in `isqrt`

Reply:
Is there a way of having the build run these types of checks for us?

Reply:
I'm sure there is - would be a really useful pr.
Eslint with the right configuration would probably work.